Add endpoint for managing recycle bin (#1920)
plone/plone.restapi: docs/source/endpoints/recycle-bin.md

> # Recycle Bin

Reviewer: Above this page title, insert required html meta stuff. See other files in this directory for examples.

```suggestion
# Recycle bin
```

> The Recycle Bin REST API provides endpoints to interact with the Plone Recycle Bin functionality.

Reviewer: ```suggestion
Plone's recycle bin functionality is managed through the `@recyclebin` endpoint.
```

## List recycle bin contents

To list all items in the recycle bin, send a GET request to the `@recyclebin` endpoint:

```http-example
GET /@recyclebin HTTP/1.1
Accept: application/json

> ```

Reviewer: This and all http example requests and responses should follow the pattern of other documentation in this directory. This allows tests to run both on the documentation and on the code. Here's example MyST markup that pulls in files as literal includes. You'll also need to create the included files.

https://github.com/plone/plone.restapi/blob/70da1de9b4f3b1aded09b5c259631bc7934c4942/docs/source/endpoints/addons.md?plain=1#L17-L30

Reviewer: See also the contributing guide at https://6.docs.plone.org/plone.restapi/docs/source/contributing/index.html#generate-documentation-examples
